src/main/java/com/autotune/experimentManager/core/ExperimentTrialHandler.java

```diff
@@ -158,12 +158,16 @@ public void startExperimentTrials() {
                 );
                 IntStream.rangeClosed(1, numberOFIterations).forEach(
                     i -> {
+                        // Setting the status as IN_PROGRESS as we are starting the trial
+                        this.experimentTrial.setStatus(EMUtil.EMExpStatus.IN_PROGRESS);
                         // Check for deployment needed to be implemented
                         deploymentHandler.initiateDeploy();
                         // Check if deployment is ready
                         EMUtil.DeploymentReadinessStatus deploymentReadinessStatus = deploymentHandler.isDeploymentReady(this.experimentTrial);
                         switch (deploymentReadinessStatus) {
                             case READY:
+                                // Setting the status to `WAITING_FOR_LOAD` as the deployment is ready
+                                this.experimentTrial.setStatus(EMUtil.EMExpStatus.WAITING_FOR_LOAD);
```

Comment:
We cannot assume that LOAD is expected here ... that depends on whether the user will provide load or if in production ... it is already there. So the status can only indicate that deployment is successful. If not it should indicate the deployment failed, so that if we retrieve the status through REST API, it should indicate the last success/failure that happened

Reply:
Can I set the status outside the flow scope? if that's the case I need to add the status based on the #502 EM Flags. I'm expecting something like this

```
Experiment trial object created - status CREATED
Based on check for start the trial - status QUEUED / IN_PROGRESS
if load check needed - status WAITING_FOR_LOAD (no status required to set in else case as next flow takes care of it)
if metric collection needed - status COLLECTING_METRICS (no status required to set in else case as next flow takes care of it)
if reached end of the flow - status COMPLETED
if any exception occured - status FAILED else status SUCCESS
```

Reply:
Should we have granular statuses? like for each sub flow result reflecting in status, if yes then I'm expecting something like this:

```
Experiment trial object created - status CREATED
Based on check for start the trial - status QUEUED / IN_PROGRESS
if deployment needed - status DEPLOYING 
    if deployment success -status DEPLOYMENT_SUCCESSFULL
    if deployment fails - status DEPLOYMENT_FAILED
if load check needed - status WAITING_FOR_LOAD (no status required to set in else case as next flow takes care of it)
      if load check failed - status LOAD_CHECK_FAILED / NO_LOAD (based on the scenerio)
      if load check success - status LOAD_CHECK_SUCCESSFULL
if metric collection needed - status COLLECTING_METRICS (no status required to set in else case as next flow takes care of it)
     if metric collection failed - status METRICS_COLLECTION_FAILED
     if metric collection success - status COLLECTED_METRICS
if reached end of the flow - status TRIAL_COMPLETED
if sending the details and getting 200 OK from analyser - status TRIAL_RESULT_SENT_SUCCESSFULLY
if sending the results fails - status TRIAL_RESULT_SEND_FAILURE
```

Reply:
Or we can also maintain a high level status as `CREATED, QUEUED, IN_PROGRESS, SUCCESS, FAILURE` and verbose statuses as `DEPLOYING, DEPLOYMENT_SUCCESSFULL, DEPLOYMENT_FAILED, WAITING_FOR_LOAD, LOAD_CHECK_FAILED, NO_LOAD, LOAD_CHECK_SUCCESSFULL, COLLECTING_METRICS, METRICS_COLLECTION_FAILED, COLLECTED_METRICS, TRIAL_RESULT_SENT_SUCCESSFULLY, TRIAL_RESULT_SEND_FAILURE`

Reply:
> Should we have granular statuses? like for each sub flow result reflecting in status, if yes then I'm expecting something like this:
> 
> ```
> Experiment trial object created - status CREATED
> Based on check for start the trial - status QUEUED / IN_PROGRESS
> if deployment needed - status DEPLOYING 
>     if deployment success -status DEPLOYMENT_SUCCESSFULL
>     if deployment fails - status DEPLOYMENT_FAILED
> if load check needed - status WAITING_FOR_LOAD (no status required to set in else case as next flow takes care of it)
>       if load check failed - status LOAD_CHECK_FAILED / NO_LOAD (based on the scenerio)
>       if load check success - status LOAD_CHECK_SUCCESSFULL
> if metric collection needed - status COLLECTING_METRICS (no status required to set in else case as next flow takes care of it)
>      if metric collection failed - status METRICS_COLLECTION_FAILED
>      if metric collection success - status COLLECTED_METRICS
> if reached end of the flow - status TRIAL_COMPLETED
> if sending the details and getting 200 OK from analyser - status TRIAL_RESULT_SENT_SUCCESSFULLY
> if sending the results fails - status TRIAL_RESULT_SEND_FAILURE
> ```

This looks good to me ... this helps in debugging as well, if anything goes wrong in between, we know exactly where we are at any point

Reply:
Sure @dinogun , As the statuses are now dependent on flow flags, I would like to get this reviewed after #502 . Will rebase and incorporate the changes and will update you on the same.